agent/metadata/server_test.go

```diff
@@ -53,173 +54,136 @@ func TestServer_Key_params(t *testing.T) {
 }
 
 func TestIsConsulServer(t *testing.T) {
```

Comment:
I rewrote this test to be easier to add cases to and folded the following optional one in.